File: Core/Entities/InmateBill.cs
using Core.Enumerations;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Core.Entities
{
    public class InmateBill : BaseEntity
    {
        [Required]
        public int Month { get; set; }
        [Required]
        public int Year { get; set; }
        public decimal BillAmount { get; set; }
        public Inmate Inmate { get; set; }
        public int InmateId { get; set; }
        public int? PaymentId { get; set; }
        public PaymentStatus PaymentStatus { get; set; } = PaymentStatus.NotPaid;
        public BillPayment BillPayment { get; set; }
        public List<BillDetail> BillItems { get; set; }
    }
}

File: WebUi/src/app/shared/models/inmateBill.ts
export interface BillItem {
    itemName: string;
    itemCategoryName: string;
    amount: number;
}

export interface InmateBill {
    id: number;
    month: string;
    year: number;
    billAmount: number;
    inmate: string;
    inmateId: number;
    paymentStatus: number;
    billItems: BillItem[];
}
